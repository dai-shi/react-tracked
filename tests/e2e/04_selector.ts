import { describe, expect, it } from 'vitest';
import puppeteer from 'puppeteer';

describe('04_selector', () => {
  const port = process.env.PORT || '8080';

  it('should work with recorded events', async () => {
    // eslint-disable-next-line import/no-named-as-default-member
    const browser = await puppeteer.launch();
    const page = await browser.newPage();
    await page.goto(`http://localhost:${port}/`);

    await page.waitForSelector(
      'body > #app > div:nth-child(2) > div > button:nth-child(2)',
    );
    await page.click(
      'body > #app > div:nth-child(2) > div > button:nth-child(2)',
    );
    expect(
      await page.evaluate(() => document.body.innerHTML),
    ).toMatchSnapshot();

    await page.waitForSelector(
      'body > #app > div:nth-child(2) > div > button:nth-child(3)',
    );
    await page.click(
      'body > #app > div:nth-child(2) > div > button:nth-child(3)',
    );
    expect(
      await page.evaluate(() => document.body.innerHTML),
    ).toMatchSnapshot();

    await page.waitForSelector(
      'body > #app > div:nth-child(3) > div > button:nth-child(2)',
    );
    await page.click(
      'body > #app > div:nth-child(3) > div > button:nth-child(2)',
    );
    expect(
      await page.evaluate(() => document.body.innerHTML),
    ).toMatchSnapshot();

    await page.waitForSelector(
      'body > #app > div:nth-child(3) > div > button:nth-child(3)',
    );
    await page.click(
      'body > #app > div:nth-child(3) > div > button:nth-child(3)',
    );
    expect(
      await page.evaluate(() => document.body.innerHTML),
    ).toMatchSnapshot();

    await page.waitForSelector(
      'body > #app > div:nth-child(5) > div:nth-child(1) > input',
    );
    await page.click(
      'body > #app > div:nth-child(5) > div:nth-child(1) > input',
    );
    expect(
      await page.evaluate(() => document.body.innerHTML),
    ).toMatchSnapshot();

    await page.type(
      'body > #app > div:nth-child(5) > div:nth-child(1) > input',
      'a',
    );
    expect(
      await page.evaluate(() => document.body.innerHTML),
    ).toMatchSnapshot();

    await page.type(
      'body > #app > div:nth-child(5) > div:nth-child(1) > input',
      'a',
    );
    expect(
      await page.evaluate(() => document.body.innerHTML),
    ).toMatchSnapshot();

    await page.type(
      'body > #app > div:nth-child(5) > div:nth-child(1) > input',
      'a',
    );
    expect(
      await page.evaluate(() => document.body.innerHTML),
    ).toMatchSnapshot();

    await page.type(
      'body > #app > div:nth-child(5) > div:nth-child(2) > input',
      'b',
    );
    expect(
      await page.evaluate(() => document.body.innerHTML),
    ).toMatchSnapshot();

    await page.type(
      'body > #app > div:nth-child(5) > div:nth-child(2) > input',
      'b',
    );
    expect(
      await page.evaluate(() => document.body.innerHTML),
    ).toMatchSnapshot();

    await page.type(
      'body > #app > div:nth-child(5) > div:nth-child(2) > input',
      'b',
    );
    expect(
      await page.evaluate(() => document.body.innerHTML),
    ).toMatchSnapshot();

    await page.type(
      'body > #app > div:nth-child(5) > div:nth-child(3) > input',
      '1',
    );
    expect(
      await page.evaluate(() => document.body.innerHTML),
    ).toMatchSnapshot();

    await page.type(
      'body > #app > div:nth-child(5) > div:nth-child(3) > input',
      '1',
    );
    expect(
      await page.evaluate(() => document.body.innerHTML),
    ).toMatchSnapshot();

    await page.type(
      'body > #app > div:nth-child(5) > div:nth-child(3) > input',
      '1',
    );
    expect(
      await page.evaluate(() => document.body.innerHTML),
    ).toMatchSnapshot();

    await browser.close();
  });
});
