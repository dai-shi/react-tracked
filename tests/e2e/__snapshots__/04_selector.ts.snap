// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`04_selector > should work with recorded events 1`] = `
"
    <div id="app"><h1>Counter</h1><div>numRendered: 14<div><span>Count: 1</span><button type="button">+1</button><button type="button">-1</button></div></div><div>numRendered: 16<div><span>Count: 1</span><button type="button">+1</button><button type="button">-1</button></div></div><h1>Person</h1><div>numRendered: 2<div>First Name:<input value=""></div><div>Last Name:<input value=""></div><div>Age:<input value="0"></div></div><div>numRendered: 4<div>First Name:<input value=""></div><div>Last Name:<input value=""></div><div>Age:<input value="0"></div></div></div>
  

"
`;

exports[`04_selector > should work with recorded events 2`] = `
"
    <div id="app"><h1>Counter</h1><div>numRendered: 22<div><span>Count: 0</span><button type="button">+1</button><button type="button">-1</button></div></div><div>numRendered: 24<div><span>Count: 0</span><button type="button">+1</button><button type="button">-1</button></div></div><h1>Person</h1><div>numRendered: 2<div>First Name:<input value=""></div><div>Last Name:<input value=""></div><div>Age:<input value="0"></div></div><div>numRendered: 4<div>First Name:<input value=""></div><div>Last Name:<input value=""></div><div>Age:<input value="0"></div></div></div>
  

"
`;

exports[`04_selector > should work with recorded events 3`] = `
"
    <div id="app"><h1>Counter</h1><div>numRendered: 30<div><span>Count: 1</span><button type="button">+1</button><button type="button">-1</button></div></div><div>numRendered: 32<div><span>Count: 1</span><button type="button">+1</button><button type="button">-1</button></div></div><h1>Person</h1><div>numRendered: 2<div>First Name:<input value=""></div><div>Last Name:<input value=""></div><div>Age:<input value="0"></div></div><div>numRendered: 4<div>First Name:<input value=""></div><div>Last Name:<input value=""></div><div>Age:<input value="0"></div></div></div>
  

"
`;

exports[`04_selector > should work with recorded events 4`] = `
"
    <div id="app"><h1>Counter</h1><div>numRendered: 38<div><span>Count: 0</span><button type="button">+1</button><button type="button">-1</button></div></div><div>numRendered: 40<div><span>Count: 0</span><button type="button">+1</button><button type="button">-1</button></div></div><h1>Person</h1><div>numRendered: 2<div>First Name:<input value=""></div><div>Last Name:<input value=""></div><div>Age:<input value="0"></div></div><div>numRendered: 4<div>First Name:<input value=""></div><div>Last Name:<input value=""></div><div>Age:<input value="0"></div></div></div>
  

"
`;

exports[`04_selector > should work with recorded events 5`] = `
"
    <div id="app"><h1>Counter</h1><div>numRendered: 38<div><span>Count: 0</span><button type="button">+1</button><button type="button">-1</button></div></div><div>numRendered: 40<div><span>Count: 0</span><button type="button">+1</button><button type="button">-1</button></div></div><h1>Person</h1><div>numRendered: 2<div>First Name:<input value=""></div><div>Last Name:<input value=""></div><div>Age:<input value="0"></div></div><div>numRendered: 4<div>First Name:<input value=""></div><div>Last Name:<input value=""></div><div>Age:<input value="0"></div></div></div>
  

"
`;

exports[`04_selector > should work with recorded events 6`] = `
"
    <div id="app"><h1>Counter</h1><div>numRendered: 38<div><span>Count: 0</span><button type="button">+1</button><button type="button">-1</button></div></div><div>numRendered: 40<div><span>Count: 0</span><button type="button">+1</button><button type="button">-1</button></div></div><h1>Person</h1><div>numRendered: 46<div>First Name:<input value="a"></div><div>Last Name:<input value=""></div><div>Age:<input value="0"></div></div><div>numRendered: 48<div>First Name:<input value="a"></div><div>Last Name:<input value=""></div><div>Age:<input value="0"></div></div></div>
  

"
`;

exports[`04_selector > should work with recorded events 7`] = `
"
    <div id="app"><h1>Counter</h1><div>numRendered: 38<div><span>Count: 0</span><button type="button">+1</button><button type="button">-1</button></div></div><div>numRendered: 40<div><span>Count: 0</span><button type="button">+1</button><button type="button">-1</button></div></div><h1>Person</h1><div>numRendered: 54<div>First Name:<input value="aa"></div><div>Last Name:<input value=""></div><div>Age:<input value="0"></div></div><div>numRendered: 56<div>First Name:<input value="aa"></div><div>Last Name:<input value=""></div><div>Age:<input value="0"></div></div></div>
  

"
`;

exports[`04_selector > should work with recorded events 8`] = `
"
    <div id="app"><h1>Counter</h1><div>numRendered: 38<div><span>Count: 0</span><button type="button">+1</button><button type="button">-1</button></div></div><div>numRendered: 40<div><span>Count: 0</span><button type="button">+1</button><button type="button">-1</button></div></div><h1>Person</h1><div>numRendered: 62<div>First Name:<input value="aaa"></div><div>Last Name:<input value=""></div><div>Age:<input value="0"></div></div><div>numRendered: 64<div>First Name:<input value="aaa"></div><div>Last Name:<input value=""></div><div>Age:<input value="0"></div></div></div>
  

"
`;

exports[`04_selector > should work with recorded events 9`] = `
"
    <div id="app"><h1>Counter</h1><div>numRendered: 38<div><span>Count: 0</span><button type="button">+1</button><button type="button">-1</button></div></div><div>numRendered: 40<div><span>Count: 0</span><button type="button">+1</button><button type="button">-1</button></div></div><h1>Person</h1><div>numRendered: 70<div>First Name:<input value="aaa"></div><div>Last Name:<input value="b"></div><div>Age:<input value="0"></div></div><div>numRendered: 72<div>First Name:<input value="aaa"></div><div>Last Name:<input value="b"></div><div>Age:<input value="0"></div></div></div>
  

"
`;

exports[`04_selector > should work with recorded events 10`] = `
"
    <div id="app"><h1>Counter</h1><div>numRendered: 38<div><span>Count: 0</span><button type="button">+1</button><button type="button">-1</button></div></div><div>numRendered: 40<div><span>Count: 0</span><button type="button">+1</button><button type="button">-1</button></div></div><h1>Person</h1><div>numRendered: 78<div>First Name:<input value="aaa"></div><div>Last Name:<input value="bb"></div><div>Age:<input value="0"></div></div><div>numRendered: 80<div>First Name:<input value="aaa"></div><div>Last Name:<input value="bb"></div><div>Age:<input value="0"></div></div></div>
  

"
`;

exports[`04_selector > should work with recorded events 11`] = `
"
    <div id="app"><h1>Counter</h1><div>numRendered: 38<div><span>Count: 0</span><button type="button">+1</button><button type="button">-1</button></div></div><div>numRendered: 40<div><span>Count: 0</span><button type="button">+1</button><button type="button">-1</button></div></div><h1>Person</h1><div>numRendered: 86<div>First Name:<input value="aaa"></div><div>Last Name:<input value="bbb"></div><div>Age:<input value="0"></div></div><div>numRendered: 88<div>First Name:<input value="aaa"></div><div>Last Name:<input value="bbb"></div><div>Age:<input value="0"></div></div></div>
  

"
`;

exports[`04_selector > should work with recorded events 12`] = `
"
    <div id="app"><h1>Counter</h1><div>numRendered: 38<div><span>Count: 0</span><button type="button">+1</button><button type="button">-1</button></div></div><div>numRendered: 40<div><span>Count: 0</span><button type="button">+1</button><button type="button">-1</button></div></div><h1>Person</h1><div>numRendered: 94<div>First Name:<input value="aaa"></div><div>Last Name:<input value="bbb"></div><div>Age:<input value="1"></div></div><div>numRendered: 96<div>First Name:<input value="aaa"></div><div>Last Name:<input value="bbb"></div><div>Age:<input value="1"></div></div></div>
  

"
`;

exports[`04_selector > should work with recorded events 13`] = `
"
    <div id="app"><h1>Counter</h1><div>numRendered: 38<div><span>Count: 0</span><button type="button">+1</button><button type="button">-1</button></div></div><div>numRendered: 40<div><span>Count: 0</span><button type="button">+1</button><button type="button">-1</button></div></div><h1>Person</h1><div>numRendered: 102<div>First Name:<input value="aaa"></div><div>Last Name:<input value="bbb"></div><div>Age:<input value="11"></div></div><div>numRendered: 104<div>First Name:<input value="aaa"></div><div>Last Name:<input value="bbb"></div><div>Age:<input value="11"></div></div></div>
  

"
`;

exports[`04_selector > should work with recorded events 14`] = `
"
    <div id="app"><h1>Counter</h1><div>numRendered: 38<div><span>Count: 0</span><button type="button">+1</button><button type="button">-1</button></div></div><div>numRendered: 40<div><span>Count: 0</span><button type="button">+1</button><button type="button">-1</button></div></div><h1>Person</h1><div>numRendered: 110<div>First Name:<input value="aaa"></div><div>Last Name:<input value="bbb"></div><div>Age:<input value="111"></div></div><div>numRendered: 112<div>First Name:<input value="aaa"></div><div>Last Name:<input value="bbb"></div><div>Age:<input value="111"></div></div></div>
  

"
`;
